maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -1022,9 +1029,9 @@ private void createDependencyResolver()
         Artifact junitDepArtifact = getJunitDepArtifact();
         return new ProviderList( new DynamicProviderInfo( null ),
                               new TestNgProviderInfo( getTestNgArtifact() ),
+                              new JUnitPlatformProviderInfo( getJunitPlatformArtifact() ),
```

Comment:
Here the order of provider-info was changed. Does it mean something important for functionality?

Reply:
Are the built-in providers documented somewhere? I couldn't find any external information.

The first auto-detected provider wins, right?

If I understand the `isApplicable()` logic correctly, the `platform` info needs to checked before the legacy JUnit (3 + 4) infos. This is due to fact, that a legacy "junit-4.12.jar" is present in both cases: the users is on 4 and wants to run JUnit 4 test cases. Or the user is on "5" and wants to Vintage Engine to execute JUnit 4 or 3 test cases.

So, when we find `org.junit.platform:junit-platform-engine` as a dependency the uses wants to launch the JUnit Platform -- perhaps with the Vintage Engine which brings along JUnit 4.x jar.